pkg/util/iptables/iptables_test.go

```diff
@@ -1049,6 +1049,8 @@ func TestRestoreAllWaitOldIptablesRestore(t *testing.T) {
 	if err == nil {
 		t.Errorf("expected failure")
 	}
+	// restore creates a file to mimic iptables lock if the runner restoreWaitFlag == 0
+	os.Remove(lockPath16x)
```

Comment:
this should probably be in a defer?

Reply:
this is the last instruction :upside_down_face:

Reply:
> this is the last instruction

yeah but it shouldn't be; if the `err == nil` case above gets hit then we wouldn't remove the file

likewise below

Reply:
Cleanup should nearly always be in a defer close to creating the resource because if anything panics in-between it will still run

Reply:
why only delete 16 and not 14 ?

Reply:
the problem is here https://github.com/kubernetes/kubernetes/blob/cea1d4e20b4a7886d8ff65f34c6d4f95efcb4742/pkg/util/iptables/iptables_linux.go#L68-L85

the file for 16 is never removed once the test ends, 14 is a socket, so I assume that the test closes it indirectly when it ends